### ISSUE TITLE
How to define conditions in ability for belongs_to, has_many association?

### ISSUE THREAD

**Issue by zhaohuic** [ryanb/cancan#759]
My two associations:

---

class Project < ActiveRecord::Base
  belongs_to :user
end

class User < ActiveRecord::Base
  include ActiveModel::MassAssignmentSecurity
  has_many :projects

  # list of attributes that lab_admin can modify
  attr_accessible :login, :cost_center, :last_name, :lab_admin, :as => :lab_admin
end

---

And in my ability class:

---

if user.admin
      can :manage, :all
elsif user.lab_admin
      can :manage, Project, :user => { :cost_center => user.cost_center}   (????)

---

Lab_admin can manage all projects that belongs_to users that have same cost_center as lab_admin. Project doesn't have cost_center attribute. It has: Name    User  ... ...

Basically, I need:
1. from user.cost_center, get the all user.id that has the same cost_center
2. from user.id, get Project.user_id
3. display

I did lots of search and cannot find a solution to this. How do I define this condition in ability? 

Thanks,
